rustls/src/server/tls12.rs

```diff
@@ -182,7 +182,29 @@ mod client_hello {
                 .provider
                 .kx_groups
                 .iter()
+                .filter(|skxg| skxg.name().key_exchange_algorithm() == Some(self.suite.kx))
```

Comment:
Not very related to here, since this part of code is following the second paragraph in [Server Behavior] in RFC 7919.

I want to discuss about the first paragraph in [Server Behavior]
```text
If a compatible TLS server receives a Supported Groups extension from
a client that includes any FFDHE group (i.e., any codepoint between
256 and 511, inclusive, even if unknown to the server), and if none
of the client-proposed FFDHE groups are known and acceptable to the
server, then the server MUST NOT select an FFDHE cipher suite.  In
this case, the server SHOULD select an acceptable non-FFDHE cipher
suite from the client's offered list.  If the extension is present
with FFDHE groups, none of the client's offered groups are acceptable
by the server, and none of the client's proposed non-FFDHE cipher
suites are acceptable to the server, the server MUST end the
connection with a fatal TLS alert of type insufficient_security(71).
```
This means some logic of choosing cipher suite need to be added in the called of current/this function: `ExpectClientHello::with_certified_key`


[Server Behavior]: https://datatracker.ietf.org/doc/html/rfc7919#section-4

Reply:
Thanks for catching this! Will fix.